WordPress/Classes/ViewRelated/Plugins/PluginViewModel.swift

```diff
@@ -2,36 +2,125 @@ import Foundation
 import WordPressFlux
 
 class PluginViewModel: Observable {
-    var plugin: Plugin {
+    var plugin: Plugin? {
         didSet {
             changeDispatcher.dispatch()
         }
     }
-    let capabilities: SitePluginCapabilities
+
+    var directoryEntry: PluginDirectoryEntry? {
```

Comment:
I think it might be problematic that the model for this are now two optionals `(Plugin?, PluginDirectoryEntry?)`. What happens if they're both `nil`?